app/main.py

```diff
@@ -1,6 +1,14 @@
-from typing import Callable
+from typing import Callable, Any
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    cache_dict = {}
+
+    def inner(*args) -> Any:
+        if args not in cache_dict:
+            cache_dict[args] = func(*args)
```

Comment:
You need to take into account both args and kwargs here. Also different functions can take same args, so let's create complex key with args, kwargs and func name
`(args, tuple(kwargs.items()), func.__name__)`

Reply:
@LLkaia What if I do it this way?
args += tuple(kwargs.values())